header/p2p/peer_tracker.go

```diff
@@ -102,3 +148,41 @@ func (p *peerTracker) peers() []*peerStat {
 	}
 	return peers
 }
+
+// gc goes through connected and disconnected peers once every gcPeriod
+// and removes:
+// * disconnected peers which have been disconnected for more than maxAwaitingTime;
+// * connected peers whose scores are less than or equal than defaultScore;
+func (p *peerTracker) gc() {
+	ticker := time.NewTicker(p.gcPeriod)
+	for {
+		select {
+		case <-p.ctx.Done():
+			p.done <- struct{}{}
+			return
+		case <-ticker.C:
+			p.Lock()
+			for id, peer := range p.disconnectedPeers {
+				if peer.pruneDeadline.Before(time.Now()) {
+					delete(p.disconnectedPeers, id)
+				}
+			}
+
+			for id, peer := range p.connectedPeers {
+				if peer.peerScore <= p.defaultScore {
+					delete(p.connectedPeers, id)
+				}
+			}
```

Comment:
We should rename connectedPeers to trackedPeers or something. Previously, we could call them so as we had 1:1 mapping of connected peers, but now we can connect to more than what we track. Because of this check and the global limit introduced in this PR